src/common/path_util.cpp

```diff
@@ -82,11 +82,22 @@ static std::filesystem::path GetBundleParentDirectory() {
 
 static auto UserPaths = [] {
 #ifdef __APPLE__
-    std::filesystem::current_path(GetBundleParentDirectory());
+    // Start by assuming the base directory is the bundle's parent directory.
+    std::filesystem::path base_dir = GetBundleParentDirectory();
+    std::filesystem::path user_dir = base_dir / "user";
+
+    // Check if the "user" directory exists in the current path:
+    if (!std::filesystem::exists(user_dir)) {
+        // If it doesn't exist, use the hardcoded path:
+        base_dir = std::filesystem::path(getenv("HOME")) / "Library/Application Support/shadPS4";
+        user_dir = base_dir / "user";
```

Comment:
I don't think the user part is actually needed here since we're already in a shadps4-specific data folder, but if you want it there then it should at least probably be using the same constant here as well.

Reply:
Yeah I was thinking that too originally, for some reason was thinking `user` folder is part of PS4's FS, nvm 🤦🏻 
Corrected.